components/engine_rocks/src/util.rs

```diff
@@ -489,16 +489,19 @@ impl CompactionFilterFactory for RangeCompactionFilterFactory {
 pub struct RangeCompactionFilter(Arc<OwnedRange>);
 
 impl CompactionFilter for RangeCompactionFilter {
-    fn featured_filter(
+    fn unsafe_filter(
         &mut self,
         _level: usize,
         key: &[u8],
         _seqno: u64,
         _value: &[u8],
         _value_type: CompactionFilterValueType,
     ) -> CompactionFilterDecision {
-        if key < self.0.start_key.as_ref() || key >= self.0.end_key.as_ref() {
-            CompactionFilterDecision::Remove
+        if key < self.0.start_key.as_ref() {
```

Comment:
`self.0.start_key` must be a data key, right?

Reply:
Should we compare with `keys::DATA_MIN_KEY` too?

Reply:
`DATA_MAX_KEY` is only needed to skip to the end. `DATA_MIN_KEY` on the other hand is unrelated to how this filter performs.